.github/workflows/format.yml

```diff
@@ -0,0 +1,41 @@
+name: Check Format
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  format:
+    runs-on: ubuntu-24.04
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+    # Checkout the repository (shallow clone)
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        submodules: recursive
+
+    # Set Git config
+    - name: Git config
+      run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+    # Run formatter
+    - name: Run clang-format
+      run: ./format
+
+    # Cancel if there's a diff
+    - name: Check status
+      run: |
+            git diff --name-only > diff.txt
+            if [ -s diff.txt ]; then
+              echo "Changes detected. Stopping workflow. Files list:"
+              cat diff.txt
+              exit 1
+            else
+              echo "Success!"
+              exit 0
+            fi
```

Comment:
if you don't care about the message, we could replace this with `git diff --name-only --exit-code`. Maybe fold it into the above like
```
- name: Check formatting
  run: ./format && git diff --name-only --exit-code
```
so you can see that "Check formatting" failed in the overview? And viewing the log for that should show the files list

Reply:
oh lol, I had no idea this was a thing

Reply:
oh I didn't realize this was in a separate workflow. Maybe separate steps rather than && is better then? up to you